docs/secondary-network.md

```diff
@@ -0,0 +1,142 @@
+  namespace: kube-system
+data:
+  antrea-agent.conf: |
+    secondaryNetwork:
+    ovsBridges: [{"bridgeName": "br-secondary", "physicalInterfaces": ["eth1"]}]
+```
+
+At the moment, Antrea supports only a single OVS bridge for secondary networks,
+and supports only a single physical interface on the bridge. The physical
+interface cannot be the Node's management interface, otherwise a Node's
+management network connectivity can be broken after `antrea-agent` creates the
+OVS bridge and moves the management interface to the bridge.
+
+## Secondary VLAN network configuration
+
+A secondary VLAN network is defined by a NetworkAttachmentDefinition CR. For
+example, the following NetworkAttachmentDefinition defines a VLAN network
+`vlan100`.
+
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: vlan100
+spec:
+  config: '{
+      "cniVersion": "0.3.0",
+      "type": "antrea",
+      "networkType": "vlan",
+      "mtu": 1500,
+      "vlan": 100,
+      "ipam": {
+        "type": "antrea",
+        "ippools": ["vlan100-ipv4", "vlan100-ipv6"]
+      }
+    }'
+```
+
+`antrea-agent` will connect Pod secondary interfaces belonging to a VLAN network
+to the secondary OVS bridge on the Node. If a non-zero VLAN is speficied in the
+network's `config`, `antrea-agent` will configure the VLAN ID on the OVS port,
+so the interface's traffic will be isolated within the VLAN. And before the
+traffic is forwarded out the Node via the secondary bridge's physical interface,
+OVS will insert the VLAN tag in the packets.
+
+A few extra notes about the NetworkAttachmentDefinition `config` fields:
+
+* `type` - must be set to `antrea`.
+* `networkType` - the only supported network type is `vlan` as of now.
+* `mtu` - defaults to 1500 if not set.
+* `vlan` - can be set to 0 or a valid VLAN ID (1 - 4094). Defaults to 0. The
+VLAN ID can also be specified as part of the spec of an IPPool referenced in the
+`ipam` section, but `vlan` in NetworkAttachmentDefinition `config` will override
+the VLAN in IPPool(s) if both are set.
+* `ipam` - it is optional. If not set, the secondary interfaces created for the
+network won't have an IP address allocated. For more information about secondary
+network IPAM configuration, please refer to the [Antrea IPAM document](antrea-ipam.md#ipam-for-secondary-network).
+
+## Pod secondary interface configuration
+
+You can create a Pod with secondary network interfaces by adding the
+`k8s.v1.cni.cncf.io/networks` annotation to the Pod. The following example Pod
+includes two secondary interfaces, one in network `vlan100` which should be
+created in the same Namespace as the Pod, the other in network `vlan200` which
+is created in Namespace `networks`.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+ name: sample-pod
+ labels:
+   app: antrea-secondary-network-demo
+ annotations:
+   k8s.v1.cni.cncf.io/networks: [vlan100, {name: vlan200, namespace: networks, interface: eth200"]
```

Comment:
the annotation value doesn't look right to me. There is an orphan `"` at the end, and if the annotation is JSON, then all the strings should be quoted

Reply:
You are right. Fixed it and added an example of a Pod with a single secondary interface.